Source/SwiftLintBuiltInRules/Rules/Style/StatementPositionRule.swift

```diff
@@ -1,217 +1,384 @@
+            case .uncuddledElse:
+                guard
+                    let (position, correctedNode) = node.uncuddledModeViolationPosition()
+                else {
+                    return super.visit(node)
                 }
-                if match.range(at: 2).length == 0 {
-                    return match
+
+                correctionPositions.append(position)
+                newNode = correctedNode
+            }
+
+            return super.visit(newNode)
+        }
+
+        override func visit(_ node: DoStmtSyntax) -> StmtSyntax {
+            guard !node.isContainedIn(regions: disabledRegions, locationConverter: locationConverter) else {
+                return super.visit(node)
+            }
+
+            var newNode = node
+
+            switch config.statementMode {
+            case .default:
+                guard let (positions, correctedNode) = node.defaultModeViolationPositions()
+                else {
+                    return super.visit(node)
                 }
-                let range1 = match.range(at: 1)
-                let range2 = match.range(at: 3)
-                let whitespace1 = contents.string.substring(from: range1.location, length: range1.length)
-                let whitespace2 = contents.string.substring(from: range2.location, length: range2.length)
-                if whitespace1 == whitespace2 {
-                    return nil
+
+                correctionPositions.append(contentsOf: positions)
+                newNode = correctedNode
+            case .uncuddledElse:
+                guard let (positions, correctedNode) = node.uncuddledModeViolationPositions()
+                else {
+                    return super.visit(node)
                 }
-                return match
+
+                correctionPositions.append(contentsOf: positions)
+                newNode = correctedNode
             }
+
+            return super.visit(newNode)
+        }
     }
+}
+
+private extension IfExprSyntax {
+    var elseBeforeIfKeyword: TokenSyntax? {
+        if
+            ifKeyword.previousToken(viewMode: .sourceAccurate)?.isElseKeyword == true,
+            let elseBeforeIfKeyword = ifKeyword.previousToken(viewMode: .sourceAccurate)
+        {
+            return elseBeforeIfKeyword
+        }
+
+        return nil
+    }
+
+    func defaultModeViolationPosition() -> (position: AbsolutePosition, newNode: IfExprSyntax)? {
+        let violationPosition: AbsolutePosition
+        var newNode = self
+
+        // Check if the node has `else` keyword
+        guard let elseKeyword else {
+            return nil
+        }
+
+        // Check if `else` doesn't have newline in leading trivia of `else`'and
+        // multiple spaces in trailing trivia of `body`'s `rightBrace`
+        guard
+            elseKeyword.leadingTrivia.isEmpty,
+            body.rightBrace.trailingTrivia.isSingleSpace == true
+        else {
+            violationPosition = body.rightBrace.endPositionBeforeTrailingTrivia
```

Comment:
Having the violation markers on the keywords (`else` and `catch`) seems more natural since their positions actually cause the rule to trigger.

Reply:
Yes, that would be different and, because of that, it's something we shouldn't do in this rewrite as it would make the reported OSS findings unusable. So let's keep this for another PR afterwards. 😉